include/deal.II/grid/reference_cell.h

```diff
@@ -588,7 +588,14 @@ class ReferenceCell
   bool
   contains_point(const Point<dim> &p, const double tolerance = 0) const;
 
-  /*
+  /**
+   * Return the Euclidean norm of the distance from @p to the unit cell squared.
+   */
+  template <int dim>
+  double
+  distance_to_unit_cell_square(const Point<dim> &p) const;
```

Comment:
Maybe `...squared()`? But why not just use the distance?

Reply:
I picked this name because we also have `Tensor::norm_square()` and `Point::distance_square()`.

I don't have a good argument either way for why to use the distance or the squared distance. It's a lot easier to minimize the squared distance and since those other two functions exist I figured I would skip the square root. Would you prefer I just implement the not-squared version?